charts/aad-pod-identity/templates/nmi-clusterrole.yaml

```diff
@@ -0,0 +1,28 @@
+{{- if .Values.rbac.enabled }}
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "aad-pod-identity.nmi.fullname" . }}
+  labels:
+    app.kubernetes.io/name: {{ template "aad-pod-identity.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: {{ .Release.Service }}
+    helm.sh/chart: {{ template "aad-pod-identity.chart" . }}
+    app.kubernetes.io/component: nmi
+rules:
+- apiGroups: ["apiextensions.k8s.io"]
+  resources: ["customresourcedefinitions"]
+  verbs: ["get", "list"]
+- apiGroups: [""]
+  resources: ["pods"]
+  verbs: ["get", "list"]
+- apiGroups: [""]
```

Comment:
I would make this optional - allows users to opt-out from granting this permission (read access to all secrets) if not required.

Reply:
This is the reason we have `get` for secrets - https://github.com/Azure/aad-pod-identity/issues/248

I think we add a conditional around it - enabled by default, user can opt out like @omerlh suggested.